pkg/limiters/global_limiter.go

```diff
@@ -60,28 +65,18 @@ func (l *GlobalLimiter) MaxAllowed(_ string) *corev1.ResourceList {
 	}
 }
 
+// ApplyLimits returns amount resources that is possible to allocate, if pool is exhasted returns zeros
 func (l *GlobalLimiter) ApplyLimits(_ string, resources *corev1.ResourceRequirements) *corev1.ResourceRequirements {
 	if l.availMem == nil || l.availCPU == nil {
 		// no limits were applied - returning as is
 		return resources
 	}
+	l.Lock()
+	defer l.Unlock()
 
 	requestCPU := resources.Requests.Cpu()
 	requestMem := resources.Requests.Memory()
 
-	if l.availCPU.IsZero() || l.availMem.IsZero() {
```

Comment:
Should we change `||` to `&&` and keep it logging the message?

Reply:
yeah, not sure yet if it'll be here.
I'm working on complete logging review.